aiohttp/client.py

```diff
@@ -325,6 +325,10 @@ def __init_subclass__(cls: Type["ClientSession"]) -> None:
         )
 
     def __del__(self, _warnings: Any = warnings) -> None:
+        if not hasattr(self, "closed"):
```

Comment:
Not sure about the fix
Though it fixes the test, but looks dirty
Probably you will be able to propose a better solution?

Reply:
Closed is a property, how can it now exist?

Reply:
What's the error it's supposed to be fixing?

Reply:
Probably easier to just revert it and let the tests run, then we can decide on any fixes.

Reply:
or order to don't touch `__del__`, I can set `self._connector = None` before raising the `ValueError`

Reply:
That would seem sensible, given that it's already initialised a new connector before that code at the moment anyway.